packages/date-time-picker/src/vaadin-date-time-picker-mixin.js

```diff
@@ -408,33 +472,53 @@ export const DateTimePickerMixin = (superClass) =>
     __changeEventHandler(event) {
       event.stopPropagation();
 
-      if (this.__dispatchChangeForValue === this.value) {
-        this._requestValidation();
-        this.__dispatchChange();
+      const isAlreadyInvalid = this.invalid;
+      const filledPickers = this.__filledPickers;
+      if (filledPickers.length === 1 && filledPickers[0].checkValidity() && !isAlreadyInvalid) {
+        // Skip if (a) only one picker is filled, (b) its value is valid by itself, and (c) the user
+        // is still interacting with the field. This is to give the user a chance to finish the input
+        // before giving him feedback. However, if the field is already in the invalid state due to
+        // a previous error, proceed to committing the value to get the error message updated.
+        return;
+      }
+
+      if (this.__hasPendingValueChange) {
+        this.__requestValueCommit();
       }
-      this.__dispatchChangeForValue = undefined;
     }
 
     /** @private */
     __openedChangedEventHandler() {
       const opened = this.__datePicker.opened || this.__timePicker.opened;
       this.style.pointerEvents = opened ? 'auto' : '';
+
+      if (!opened && this.__outsideClickInProgress) {
+        this.__requestValueCommit();
+      }
     }
 
     /** @private */
     __addInputListeners(node) {
       node.addEventListener('change', this.__changeEventHandler);
+      node.addEventListener('unparsable-change', this.__changeEventHandler);
       node.addEventListener('value-changed', this.__valueChangedEventHandler);
       node.addEventListener('opened-changed', this.__openedChangedEventHandler);
     }
 
     /** @private */
     __removeInputListeners(node) {
       node.removeEventListener('change', this.__changeEventHandler);
+      node.removeEventListener('unparsable-change', this.__changeEventHandler);
       node.removeEventListener('value-changed', this.__valueChangedEventHandler);
       node.removeEventListener('opened-changed', this.__openedChangedEventHandler);
     }
 
+    /** @private */
+    __requestValueCommit() {
+      this._requestValidation();
```

Comment:
It seems that `this._requestValidation()` could be now called from `this.__commitPendingValueChange()`, removing the need for this extra method.

Reply:
Done.